tests/cnf/ran/internal/ranhelper/version.go

```diff
@@ -0,0 +1,211 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// InitializeTalmVersion initializes the version of the TALM operator from the hub cluster.
+func InitializeTalmVersion() error {
+	var err error
+
+	ranparam.TalmVersion, err = getOperatorVersionFromCsv(
+		raninittools.HubAPIClient, ranparam.TalmOperatorHubNamespace, ranparam.OpenshiftOperatorNamespace)
+
+	return err
+}
+
+// InitializeVersions initializes the versions of ACM, TALM, and ZTP from the hub cluster.
+func InitializeVersions() error {
+	var err error
+
+	err = InitializeTalmVersion()
+	if err != nil {
+		return err
+	}
+
+	ranparam.AcmVersion, err = getOperatorVersionFromCsv(
+		raninittools.HubAPIClient, ranparam.AcmOperatorName, ranparam.AcmOperatorNamespace)
+	if err != nil {
+		return err
+	}
+
+	ranparam.ZtpVersion, err = getZtpVersionFromArgoCd(
+		raninittools.HubAPIClient, ranparam.OpenshiftGitopsRepoServer, ranparam.OpenshiftGitops)
+
+	return err
+}
+
+// IsVersionStringInRange checks if a version string is between a specified min and max value, inclusive. All the string
+// inputs to this function should be dot separated positive intergers, e.g. "1.0.0" or "4.10". Each string input must be
+// at least two dot separarted integers but may also be 3 or more, though only the first two are compared.
+func IsVersionStringInRange(version, minimum, maximum string) (bool, error) {
+	versionValid, versionDigits := validateInputString(version)
+	minimumValid, minimumDigits := validateInputString(minimum)
+	maximumValid, maximumDigits := validateInputString(maximum)
+
+	if !minimumValid {
+		// Only accept invalid empty strings
+		if minimum != "" {
+			return false, fmt.Errorf("invalid minimum provided: '%s'", minimum)
+		}
+
+		// Assume the minimum digits are [0,0] for later comparison
+		minimumDigits = []int{0, 0}
+	}
+
+	if !maximumValid {
+		// Only accept invalid empty strings
+		if maximum != "" {
+			return false, fmt.Errorf("invalid maximum provided: '%s'", maximum)
+		}
+
+		// Assume the maximum digits are [math.MaxInt, math.MaxInt] for later comparison
+		maximumDigits = []int{math.MaxInt, math.MaxInt}
+	}
+
+	// If the version was not valid then we need to check the min and max
+	if !versionValid {
+		// If no min or max was defined then return true
+		if !minimumValid && !maximumValid {
+			return true, nil
+		}
+
+		// Otherwise return whether the input maximum was an empty string or not
+		return maximum == "", nil
+	}
+
+	// Otherwise the versions were valid so compare the digits
+	for i := 0; i < 2; i++ {
+		// The version bit should be between the minimum and maximum
+		if versionDigits[i] < minimumDigits[i] || versionDigits[i] > maximumDigits[i] {
+			return false, nil
+		}
+	}
+
+	// At the end if we never returned then all the digits were in valid range
+	return true, nil
+}
+
+// validateInputString validates that a string is at least two dot separated nonnegative integers.
+func validateInputString(input string) (bool, []int) {
```

Comment:
Same here, a unit test to lock in the functionality would be good for this one.

Reply:
This could also be a regex to validate the structure of the string.

Something like `regexp.MatchString(input, "[0-9]+.[0-9]+")`.